phoenix-core-client/src/main/java/org/apache/phoenix/index/IndexMaintainer.java

```diff
@@ -785,7 +792,11 @@ public byte[] buildRowKey(ValueGetter valueGetter, ImmutableBytesWritable rowKey
                 	dataColumnType = expression.getDataType();
                 	dataSortOrder = expression.getSortOrder();
                     isNullable = expression.isNullable();
-                	expression.evaluate(new ValueGetterTuple(valueGetter, ts), ptr);
+                    if (expression instanceof PartitionIdFunction) {
```

Comment:
Add check for ensuring that the partition id expression is the prefix of the index table row key (by Tanuj)